### PR TITLE
Ticket 5662: Added tests for Kepco magnet calibration

### PR DESCRIPTION
**Pull request by DominicOram**
See https://github.com/ISISComputingGroup/IBEX/issues/5662

* Moved generic setting of calibration files out of eurotherm and into utils
* Added tests for this in the kepco

### CODE REVIEW

File: tests/kepco.py [hunk 1]
```diff
@@ -139,3 +141,44 @@ def test_GIVEN_psu_in_local_mode_WHEN_setpoint_is_sent_THEN_power_supply_put_int
...
+    def test_GIVEN_calibration_of_none_WHEN_current_read_THEN_field_equal(self):
+        test_current = 100
+        expected_field = 100
+
+        with use_calibration_file(self.ca, "None.txt"):
+            self._write_current(test_current)
+            self.ca.assert_that_pv_is("FIELD", expected_field)
+
+    @skip_if_recsim("Calibration lookup does not work in recsim")
+    def test_GIVEN_calibration_of_none_WHEN_current_read_THEN_field_equal(self):
```
Comment (rerpha) [inline tests/kepco.py:176]: Perhaps a little bit over the top, but could this be parametrized with the file name and a tuple of the expected current and fields? or list of tuples, then you can test multiple values

File: utils/calibration_utils.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+def set_calibration_file(channel_access, filename):
+    """
+    Sets a calibration file. Retries if it didn't set properly first time.
+    """
+    max_retries = 10
+
+    for _ in range(max_retries):
+        channel_access.set_pv_value("CAL:SEL", filename)
+        channel_access.assert_that_pv_alarm_is("CAL:SEL", channel_access.Alarms.NONE)
+        time.sleep(3)
```
Comment (rerpha) [inline utils/calibration_utils.py:14]: can we process the pv here instead of manually sleeping?

File: utils/calibration_utils.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+
+def set_calibration_file(channel_access, filename):
+    """
+    Sets a calibration file. Retries if it didn't set properly first time.
+    """
+    max_retries = 10
+
+    for _ in range(max_retries):
+        channel_access.set_pv_value("CAL:SEL", filename)
```
Comment (rerpha) [inline utils/calibration_utils.py:12]: I think "CAL:SEL" should be moved out into a constant here in case we ever change the pv name

**Review by rerpha — COMMENTED** [review]
tests look good, some minor comments. We should make sure the test framework copies in the `None.txt` file into `config/calib` for testing

File: utils/calibration_utils.py (hunk 2 above)
Comment (DominicOram) [reply utils/calibration_utils.py:14]: Good suggestion but in this case I'm not 100% sure where the slow down is coming from so I think we'll err on the side of stability over speed.

**Review by rerpha — APPROVED** [review]

**Pull request merged**